src/server/debug/server/server.go

```diff
@@ -211,18 +214,22 @@ func (s *debugServer) DumpV2(request *debug.DumpV2Request, server debug.Debug_Du
 	if request == nil {
 		return errors.New("nil debug.DumpV2Request")
 	}
-	if request.System != nil {
-		var apps []*debug.App
-		// fill the pod info for all the pods referenced in request.System Apps
-		apps = append(apps, request.System.Binaries...)
-		apps = append(apps, request.System.Describes...)
-		apps = append(apps, request.System.Logs...)
-		apps = append(apps, request.System.LokiLogs...)
-		apps = append(apps, request.System.Profiles...)
-		if err := s.fillApps(server.Context(), apps); err != nil {
-			return err
+	var apps []*debug.App
+	for _, l := range [][]*debug.App{
+		request.GetSystem().GetBinaries(),
+		request.GetSystem().GetDescribes(),
+		request.GetSystem().GetLogs(),
+		request.GetSystem().GetLokiLogs(),
+		request.GetSystem().GetProfiles(),
+	} {
+		if len(l) > 0 {
+			apps = append(apps, l...)
```

Comment:
This could just be:

```go
var apps []*debug.App
apps = append(apps, request.GetSystem().GetBinaries()...)
apps = append(apps, request.GetSystem().GetDescribess()...)
apps = append(apps, request.GetSystem().GetLogs()...)
apps = append(apps, request.GetSystem().GetLokiLogs()...)
apps = append(apps, request.GetSystem().GetProfiles()...)
```